### PR TITLE
[security-agent] Allow auth_token to be read by the datadog agent

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

Set the datadog user as the owner of the API auth token on Unix and
set ACL on Windows.

### Motivation

When started before the agent, the security agent could create an auth_token
file. As it runs as root, the Datadog agent could not read it.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/config_windows.go [hunk 1]
```diff
@@ -28,7 +28,11 @@ var (
 const ServiceName = "DatadogAgent"
 
 func osinit() {
-	pd, err := winutil.GetProgramDataDir()
```
Comment (brycekahle) [inline pkg/config/config_windows.go:31]: Is this correct? It seems like you would want this line.

Comment (lebauce) [reply pkg/config/config_windows.go:31]: That was a test that I committed by mistake. Thanks for the catch !

**Review by brycekahle — APPROVED** [review]

File: pkg/dogstatsd/server.go [hunk 2]
```diff
@@ -327,7 +325,6 @@ func NewServer(aggregator *aggregator.BufferedAggregator, extraTags []string) (*
 		eolTerminationUDP:         eolTerminationUDP,
 		eolTerminationUDS:         eolTerminationUDS,
 		eolTerminationNamedPipe:   eolTerminationNamedPipe,
-		telemetryEnabled:          telemetry_utils.IsEnabled(),
```
Comment (djmitche) [inline pkg/dogstatsd/server.go:330]: Removing this unused global seems non-problematic, but unrelated.  Was this mixed in by mistake?

**Review by djmitche — APPROVED** [review]
Could you add some detailed QA steps to ensure that the affected agents all create their auth_token files correctly?

File: pkg/dogstatsd/server.go (hunk 2 above)
Comment (lebauce) [reply pkg/dogstatsd/server.go:330]: It seems it was part of my stash. This is deadcode anyway but I can create a PR if you prefer

Comment (djmitche) [reply pkg/dogstatsd/server.go:330]: Meh, might as well leave it in :)

**Pull request merged**